### DISCUSSION TITLE
No dataset specified for DDname

### DISCUSSION THREAD

**Discussion by j-m-watkins** [IBM/zoau-samples#65] [Q&A] [answered]
When using mvscmd, how do I code a DDname without a DSname. This would be the JCL statement:

`//MYDD  DD SPACE=(CYL,1)`

Why I run a JCL job, it looks like SMS does its magic and allocates a system temporary dataset. In ZOAU I get a syntax error if I try this:

`--MYDD=,primary=1C`

`BGYSC0212E No dataset name specified for DDname MYDD`

**Comment by Degilio** [comment, marked as answer]
When you submit JCL, JES handles the creation of temporary datasets. Because you don't have the support of JES, You need to define the dataset or file yourself. You are also responsible for deleting those temporary datasets. 
It sounds like I need to create sample code to manage this for people.

**Reply by j-m-watkins** [reply]
It would be nice if ZOAU generated the temporary files automagically. Otherwise we have to:
`
dsname=$(mvstmp $(hlq))
... --MYDD=$dsname,NEW,....
drm $dsname 
`
Which is quite painful :-)

If we could do this:
`... --MYDD=,primary=1C,type=SEQ`
and ZOAU automatically created and deleted a temporary MVS sequential file based on JES conventions.

**Comment by Degilio** [comment]
How would ZOAU know when you are done with the file? It seems to me that we could build a create temporary file and an erase a temporary file, but because I don't know when things are done, the cleanup is still going to be part of your responsibility. Heck - I could easy create a routine that creates temp files, and one that erases them for you, but you still have to call them. I can try to build the content you are asking for but I can't think of a way to keep track of it for you. 
To be fair - I do not work for ZOAU and I can't speak for them. I can tell you how I have been doing things.

**Reply by j-m-watkins** [reply]
Ah, OK. So because JES knows you are running a JCL job it can manage dataset allocation and cleanup. Each mvscmd call to ZOAU is a single step in a "job", so the script writer is now JES :-)
